src/bin/pg_autoctl/pidfile.c

```diff
@@ -510,8 +510,11 @@ wait_for_pid_to_exit(const char *pidfile, int timeout, pid_t *pid)
 	{
 		if (read_pidfile(pidfile, pid))
 		{
-			log_info("An instance of pg_autoctl is running with PID %d, "
-					 "waiting for it to stop.", *pid);
+			if (timeout > 0)
+			{
+				log_info("An instance of pg_autoctl is running with PID %d, "
+						 "waiting for it to stop.", *pid);
+			}
```

Comment:
Here too some comments about our intention with the timeout-zero special case would be nice to have.